music_assistant/providers/pandora/provider.py

```diff
@@ -0,0 +1,400 @@
+            return web.Response(status=400, text="Invalid track_num")
+
+        # Calculate which fragment and which track within that fragment
+        fragment_idx = track_num // TRACKS_PER_FRAGMENT
+        track_idx = track_num % TRACKS_PER_FRAGMENT
+
+        self.logger.debug(
+            "Stream request: track %d = fragment %d, track %d",
+            track_num,
+            fragment_idx,
+            track_idx,
+        )
+
+        try:
+            # Fetch the fragment (cached automatically by decorator)
+            fragment = await self._get_fragment_data(station_id, fragment_idx)
+
+            # Get the tracks from the fragment
+            tracks = fragment.get("tracks", [])
+            if track_idx >= len(tracks):
+                self.logger.error(
+                    "Track index %d out of range (fragment has %d tracks)",
+                    track_idx,
+                    len(tracks),
+                )
+                return web.Response(status=404, text="Track not found in fragment")
+
+            track = tracks[track_idx]
+            audio_url = track.get("audioURL")
+
+            if not audio_url:
+                self.logger.error("No audio URL in track data")
+                return web.Response(status=404, text="No audio URL available")
+
+            self.logger.debug(
+                "Redirecting to audio URL for track %d (%s - %s)",
+                track_num,
+                track.get("artistName", "Unknown"),
+                track.get("songTitle", "Unknown"),
+            )
+
+            # Redirect to the actual audio URL
+            return web.Response(status=302, headers={"Location": audio_url})
+
+        except Exception as err:
+            self.logger.error(
+                "Error handling stream request for track %d: %s",
+                track_num,
+                err,
+            )
+            return web.Response(status=500, text=f"Stream error: {err}")
+
+    async def search(
+        self,
+        search_query: str,
+        media_types: list[MediaType],
+        limit: int = 25,
+    ) -> SearchResults:
+        """Search library radio stations by name."""
+        # We can't search Pandora's API without the legacy endpoints,
+        # but we can search the user's existing library stations
+        if MediaType.RADIO not in media_types:
+            return SearchResults()
+
+        results: list[Radio] = []
+
+        async for station in self.get_library_radios():
+            if compare_strings(station.name, search_query):
+                results.append(station)
+                if len(results) >= limit:
+                    break
+
+        return SearchResults(radio=results)
+
+    async def browse(self, path: str) -> list[Radio]:
+        """Browse radio stations."""
+        # For now, just return all stations like get_library_radios
+        # Could be enhanced with categories/genres in the future
+        return [station async for station in self.get_library_radios()]
```

Comment:
this can be left out if you dont support this yet. So the default implementation can be used.

Reply:
yeah, it will only work after the first sync was completed.

Reply:
After the sync I see the stations in the radio view but there is nothing when I browse to Pandora? This is after I see this in the log
2025-10-28 11:40:25.015 INFO (MainThread) [music_assistant.music] Sync task for Pandora/radios completed

Reply:
but you did implement the get_library_radios ?

Reply:
Yes on line 196?

Reply:
still you need to drop this. it should not be needed